File: First-Appraisal/src/Java/Scores.java
package Java;

/**
 * @Author ChenJialong
 * @Date 2021/10/5 19:07
 * @Description
 * 第一题（Java 基础语法）期末到了，你用 java 程序帮助老师统计学生们的成绩。
 *  * 要求：设计两个静态方法，
 *  * ·方法 1 统计学生们的平均成绩，接受一个学生成绩的(int[])整数数组 scores，返
 *  * 回平均分数(double)；
 *  * 示例：给定 scores = {59, 64, 86, 48, 76}，平均成绩 average = 66.6
 *  * ·方法 2 同样接受学生成绩的(int[])整数数组 scores，找出分数最高的同学，返回
 *  * 对应下标(int)。
 *  * 示例：给定 scores = {59, 64, 86, 48, 76}，返回分数最高同学(86)的下标:2
 *  * @Since version-1.0
 */
public class Scores {
    static public int indexOfMax(int[] scores){
        int max=-1;
        int index=-1;
        for (int i = 0; i < scores.length; i++) {
            if(scores[i]>max){
                max=scores[i];
                index=i;
            }

        }
        return index;
    }

    static public double average(int[] scores){
        double sum=0;
        for (int i = 0; i < scores.length; i++) {
            sum+=scores[i];
        }
        return sum/scores.length;
    }

}

File: first-Appraisal/src/Kotlin/Scores.kt
package Kotlin
fun main(){
    //该文件本身既是实现又是测试类。
    var arr= listOf<Int>(59, 64, 86, 48, 76);
    println("平均成绩为"+average(arr));
    println("最高成绩所在下标为"+ maxIndex(arr));

}
fun average(arr: List<Int>):Double{
    var sum=0;
    for(num in arr){
        sum+=num;
    }
    var len:Double=arr.size*1.0;
    return sum/len;

}
fun maxIndex(arr: List<Int>):Int{
    var max=-1;
    var maxIndex=-1;
    for(i in 0..arr.size-1){
        if(arr[i]>max){
            max=arr[i]
            maxIndex=i
        }
    }
    return maxIndex;

}